tls/s2n_resume.c

```diff
@@ -868,18 +872,21 @@ int s2n_decrypt_session_ticket(struct s2n_connection *conn, struct s2n_stuffer *
     POSIX_GUARD(s2n_stuffer_skip_write(&state_stuffer, state_blob_size));
     POSIX_GUARD_RESULT(s2n_deserialize_resumption_state(conn, &from->blob, &state_stuffer));
 
-    uint64_t now = 0;
-    POSIX_GUARD_RESULT(s2n_config_wall_clock(conn->config, &now));
+    if (s2n_connection_get_protocol_version(conn) >= S2N_TLS13) {
+        return S2N_SUCCESS;
+    }
 
-    /* If the key is in decrypt-only state, then a new key is assigned
-     * for the ticket.
+    /* A new key is assigned for the ticket if the key is in decrypt-only state, 
+     * a key in encrypt-decrypt state is available, and TLS 1.2 is being used.
      */
+    uint64_t now = 0;
+    POSIX_GUARD_RESULT(s2n_config_wall_clock(conn->config, &now));
+    bool is_encrypt_decrypt_key_available = false;
+    POSIX_GUARD_RESULT(s2n_config_is_encrypt_decrypt_key_available(conn->config, &is_encrypt_decrypt_key_available));
```

Comment:
Nit: You can keep this inside of the if-statement, it doesn't need to be called if the first check fails.